packages/top-app-bar/src/mwc-top-app-bar-base.ts

```diff
@@ -139,12 +121,49 @@ export class TopAppBarBase extends BaseElement {
     }
   }
 
+  protected handleTargetScroll = () => {
+    this.mdcFoundation.handleTargetScroll();
+  }
+
+  protected handleResize = () => {
+    this.mdcFoundation.handleWindowResize();
+  }
+
+  protected handleNavigationClick = () => {
+    this.mdcFoundation.handleNavigationClick();
+  }
+
+  protected registerListeners() {
+    this._navIconSlot.addEventListener('click', this.handleNavigationClick);
```

Comment:
Can the click and scroll listeners be in the template? If so, I think that's a cleaner place to do that.

Reply:
I moved the click listener in. I had it here before because I was running into a dumb issue where the foundation was not defined yet, but all worked if I just moved that call out into another function. scrollTarget may change from window to an HTML element, so I decided to keep that there.